WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -209,10 +209,11 @@ private void unpackMediaIntent(@NonNull Intent intent) {
             for (MediaModel media : mediaList) {
                 MediaModel localMedia = mMediaStore.getMediaWithLocalId(media.getId());
                 boolean notUploadedYet = localMedia != null
-                                         && MediaUploadState.fromString(localMedia.getUploadState())
-                                            != MediaUploadState.UPLOADED;
+                                         && (localMedia.getUploadState() == null
+                                             || MediaUploadState.fromString(localMedia.getUploadState())
+                                                != MediaUploadState.UPLOADED);
                 if (notUploadedYet) {
-                    toBeUploadedMediaList.add(localMedia);
+                    toBeUploadedMediaList.add(media);
```

Comment:
@ashiagr Is this change intentional? Why do we want to pass a possibly out-dated item instead of an item which was just loaded from the db? Are there some fields which are not being saved into the db and we need them?

Reply:
Yes this was intentional. For the site icon, attributes in the DB were set as null and upload didn't succeed - it failed with the message:
`MediaUploadHandler > Unexpected state, site is null. Skipping this request.`

![null_state](https://user-images.githubusercontent.com/1405144/99945251-50475e00-2d9a-11eb-8da0-09e0a44b79c5.png)

Since we already skipped media in uploaded state, it felt fine to add original media for the other states.